### PR TITLE
VZ-4042 Introduction to VZ Components Version during VZ Install (via BOM)

### PR DESCRIPTION
**Pull request by absBitm**
Introducing vz component version during vz install
extension pr of [3855](https://github.com/verrazzano/verrazzano/pull/3855)

### CODE REVIEW

File: pkg/bom/testdata/verrazzano-bom.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     },
     {
       "name": "cert-manager",
```
Comment (pfmackin) [inline pkg/bom/testdata/verrazzano-bom.json:45]: You should add a version for every component

File: platform-operator/verrazzano-bom.json [hunk 2]
```diff
@@ -351,6 +359,7 @@
     },
     {
       "name": "kiali-server",
+      "version": "0.0.0",
```
Comment (pfmackin) [inline platform-operator/verrazzano-bom.json:362]: This is not a valid version

File: platform-operator/controllers/verrazzano/controller.go [hunk 3]
```diff
@@ -670,6 +684,13 @@ func (r *Reconciler) updateComponentStatus(compContext spi.ComponentContext, mes
 	// Set the state of resource
 	componentStatus.State = checkCondtitionType(conditionType)
 
+	// Set the version of component
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/controller.go:687]: This should only be done post-install/update and post-upgrade.  Isn't this function called during those operations?

File: platform-operator/controllers/verrazzano/controller.go [hunk 4]
```diff
@@ -670,6 +684,13 @@ func (r *Reconciler) updateComponentStatus(compContext spi.ComponentContext, mes
...
+	if bomFile, err := r.getBOM(); err != nil {
+		if component, er := bomFile.GetComponent(componentName); er != nil {
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/controller.go:689]: Don't update components that are disabled

File: pkg/bom/testdata/verrazzano-bom.json [hunk 5]
```diff
@@ -43,6 +43,7 @@
...
+      "version": "v1.7.1",
```
Comment (michael-cico) [inline pkg/bom/testdata/verrazzano-bom.json:46]: Looks like we're missing some versions for components, e.g., NGINX, Istio?

File: platform-operator/verrazzano-bom.json [hunk 6]
```diff
@@ -43,6 +43,7 @@
     },
     {
       "name": "cert-manager",
+      "version": "v1.7.1",
```
Comment (michael-cico) [inline platform-operator/verrazzano-bom.json:46]: Looks like we're missing some versions for components, e.g., NGINX, Istio, external-dns?

File: platform-operator/controllers/verrazzano/controller.go [hunk 7]
```diff
@@ -670,6 +684,15 @@ func (r *Reconciler) updateComponentStatus(compContext spi.ComponentContext, mes
 	// Set the state of resource
 	componentStatus.State = checkCondtitionType(conditionType)
 
+	// Set the version of component
+	if conditionType == installv1alpha1.CondInstallComplete || conditionType == installv1alpha1.CondUpgradeStarted {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/controller.go:688]: I think from what @pfmackin said we should set the version on `CondUpgradeComplete`, not `CondUpgradeStarted`

File: platform-operator/verrazzano-bom.json (hunk 6 above)
Comment (absBitm) [reply platform-operator/verrazzano-bom.json:46]: yes for the first pass, we are only focusing on the component which includes all of the Verrazzano components

**Review by michael-cico — APPROVED** [review]

**Pull request merged**